src/asl.ts

```diff
@@ -5526,28 +5253,1768 @@ export namespace ASLGraph {
   }
 }
 
-export namespace ASL {
-  export function isTruthy(v: string): Condition {
-    return and(
-      isPresentAndNotNull(v),
-      or(
-        and(isString(v), not(stringEquals(v, ""))),
-        and(isNumeric(v), not(numericEquals(v, 0))),
-        and(isBoolean(v), booleanEquals(v, true))
-      )
+/**
+ * Operations that can help "optimize" an ASL state machine.
+ */
+namespace ASLOptimizer {
```

Comment:
Maybe this can be in its own file?

Reply:
ha, it could :) though there will be some circular references if the SFN types do not move too. leave it for now?

Reply:
On second thought, this file may be why ts is crashing for me. Will break it all up.

Reply:
https://github.com/functionless/functionless/pull/437